.ci/xenial.jenkinsfile

```diff
@@ -166,9 +186,30 @@ node('Linux') {
         echo 'Run tests that check binaries work in compatible images'
     }
 
-    // TODO: If master branch
-    stage('Upload images') {
-        echo 'Upload images'
-    }
+
+    //if (isMaster) {
+        // If it was master and everything is ok up to here
+        stage('Upload images') {
+            List images = ["base-${distro}"]
+            gccVersions.each { v ->
+                String versionMajor = getVersionMajor(v)
+                images.add("gcc${versionMajor}-${distro}")
+                images.add("gcc${versionMajor}-${distro}-jenkins")
+            }
+            clangVersions.each { v ->
+                String versionMajor = getVersionMajor(v)
+                images.add("clang${versionMajor}-${distro}")
+                images.add("clang${versionMajor}-${distro}-jenkins")
+            }
+
+            images.each { image ->
+                String builtImage = "${dockerUsername}/${image}:${branchName}-${commit}"
+                String uploadImage = "${originalUsername}/${image}:${conanVersion}"
```

Comment:
Please, @uilianries , double-check this is the expected final name for the images we are building

Reply:
I would consider using `git rev-parse --short HEAD` at line 41. It's very rare a collision, and is more readable

Reply:
👍  np

Reply:
Anyway, take into account that images tagged that way are not going to be used by humans, they are only used to save the state so we can hydrate the cache afterward.